tests/snapshots/snapshot_tests__workspace_no_lib_targets_error.snap

```diff
@@ -0,0 +1,7 @@
+---
+source: tests/snapshot_tests.rs
+expression: "&output"
+---
+exit_code: 1
+stderr: "error: no crates with library targets selected, nothing to semver-check\nnote: only library targets contain an API surface that can be checked for semver\nnote: skipped the following crates since they have no library target: bin_only\n"
```

Comment:
This isn't the most readable output: the explicit `\n` are quite annoying, as are the extremely long line lengths.

Consider individually snapshotting the stderr and stdout into separate files that are stored verbatim, like rustc does with its ui tests? For example: https://github.com/rust-lang/rust/blob/fd8d6fbe505ecf913f5e2ca590c69a7da2789879/tests/ui/nll/issue-50716.stderr

Reply:
Ron output:

```ron
CommandOutput(
  exit_code: Some(1),
  stderr: "error: no crates with library targets selected, nothing to semver-check\nnote: only library targets contain an API surface that can be checked for semver\nnote: skipped the following crates since they have no library target: bin_only\n",
  stdout: "",
)
```

I think we can use the default `insta::assert_snapshot!` on the stdout and stderr, which will use their `Display` impls, which should print new lines.

Reply:
Consider doing a hybrid approach. Make the `CommandOutput` struct contain three fields: `stdout`, `stderr`, and "everything else" inside a single struct that can be ron-serialized and snapshotted that way.

Separately, consider looking at the "redactions" and "filters" features of `insta` to remove information that will be a source of non-determinism: https://insta.rs/docs/redactions/

Reply:
It looks like `insta` provides the `insta-cmd` crate which does exactly what we want.  I refactored to use that.

Reply:
Wow, TIL! Nice find, that looks quite good ✨